gh-106318: Add an example for `str.capitalize()` (#134272)
python/cpython: Doc/library/stdtypes.rst

|                          +-------------------------------------------+---------------------------------------------------+
|                          |  :meth:`str.casefold`                     |                                                   |
|                          +-------------------------------------------+---------------------------------------------------+

> |                          |  :meth:`str.`                   |  :meth:`bytes.capitalize`                         |

Reviewer: This seems accidental.

Reviewer 2: Sorry, it was. Anyway, this commit should be reverted.

.. method:: str.capitalize()

   Return a copy of the string with its first character capitalized and the
   rest lowercased. For example::

      >>> 'PYTHON IS AMAZING'.capitalize()
      'Python is amazing'
      >>> 'ǋemačka Starts With a non-english Digraph'.capitalize()

>       'ǋemačka starts with a non-english digraph'

Reviewer: Maybe an example with a bigger visual difference?

Maybe
```
'Ǉjubljana starts With a non-english digraph'.capitalize()
```
since it is then `ǈ`

Author: @StanFromIreland , oooh I see what you mean, thank you.

>       'ǈjubljana starts with a non-english digraph'

Reviewer 3: ```suggestion
      >>> 'Ǉjubljana starts With a non-English digraph'.capitalize()
      'ǈjubljana starts with a non-english digraph'
```

Reviewer 3: It doesn't look like the word Ljubljana is written with a digraph though?

https://sl.wikipedia.org/wiki/Ljubljana

Is there a word we can use that does?

(Plus we effectively have an L and two Js in "Ǉjubljana".)

Reviewer 2: My initial suggestion was ǋemačka[0], the word used in the bug report to fix capitalize[1].

[0] https://github.com/python/cpython/pull/105670/files#diff-08a31a70dd1f6d97aa8dacdce77db4de04c700d9949be1af611a595186aad5b3R1628

[1] https://bugs.python.org/msg339568

Reviewer 3: Right, so:
```suggestion
      >>> 'ǋemačka starts with a non-English digraph'.capitalize()
      'ǋemačka starts with a non-english digraph'
```

But I'm not sure if the digraph example adds much here, so shall we skip it?

```suggestion
```

Reviewer 2: > But I'm not sure if the digraph example adds much here, so shall we skip it?

I thought it could be useful when I saw the Changed in version 3.8 note. In general, I also think useful to have examples with non-English or non-ASCII characters.

Reviewer 2: Something was wrong here. I intented to commit the suggestion with 'ǋemačka', but it disappeared. It was applied the suggestion from 23th March.

>       >>> 'Ǉjubljana starts With a non-English digraph'.capitalize()
>       'ǈjubljana starts with a non-english digraph'

Reviewer 3: Probably just a mis-click, let's try again (and with "non-ASCII"):

```suggestion
      >>> 'ǋemačka starts With a non-ASCII digraph'.capitalize()
      'ǋemačka starts with a non-ascii digraph'
```

Reviewer 2: It worked. Thank you.

Reviewer 3: Hmm, I noticed my phone (Android with Chrome) renders "ǋ" as a box:

<img width="1080" height="576" alt="image" src="https://github.com/user-attachments/assets/5ceaaa85-b710-4fd8-9b4d-99886a54260c" />

I wonder if there's a more universally supported character?

Reviewer 2: I've tried every digraph listed here: https://en.wikipedia.org/wiki/Digraph_(orthography)#In_Unicode

And all of them have failed. I think doesn't a more universally supported character. But, the problem is only on the interactive shell, on the regular text it is ok.

So, we need to update the font-family. But it is a docs issue or a Sphinx issue?

As I could see, we have:
Interactive shell font-family: Menlo, Consolas, Monaco, Liberation Mono, Lucida Console, monospace;

Regular text font-family: -apple-system, BlinkMacSystemFont, avenir next, avenir, segoe ui, helvetica neue, helvetica, Cantarell, Ubuntu, roboto, noto, arial, sans-serif;

![WhatsApp Image 2025-10-01 at 17 08 25](https://github.com/user-attachments/assets/0cf9c0f1-bfe8-4220-b75a-a020742d98fb)

Reviewer 2: I've reverted the experimental commits.
